### PR TITLE
Make BoneAttachment and Skeleton3D naming consistent

### PR DESCRIPTION
**Pull request by lyuma**
Make BoneAttachment3D and Skeleton3D names consistent.
BoneAttachment3D nodes are now named as their bone name.
Resolves cases where BoneAttachment3D nodes will have arbitrary names like BoneAttachment3D6
Reserve the name "Skeleton3D" and use this name for all generated Skeleton3D nodes.

BoneAttachment3D nodes are now named as their bone name. Resolves cases where BoneAttachment3D nodes will have arbitrary names like BoneAttachment3D6

Reserve the name "Skeleton3D" and use it for all Skeleton3D nodes.

This change will break existing imported scenes with more than one skeleton and/or bone attachments.

Tested with the godot-tests repository. (with #72158 )
skeleton_pr_comparison

### CODE REVIEW

**Comment by lyuma** [comment]
> fire: there's a risk that two sibling skeletons might not be merged and that will not have a unique name
```
Parent
  - Skeleton3D
      - Hips
  - Skeleton3D
      - Hips
```
For now, if this hypothetical case happens, one of the skeletons may be assigned a unique name based on the `true` argument of `add_child`. If this causes problems, we may need to explore other ways to deal with sibling skeletons.

Another point I wanted to make about Skeleton3D naming. The reason I'm adamantly in favor of merging this change now, is because the skeletons having arbitrary numbering will create a maintainability problem for the entire 4.x cycle. For example, you'll see in the above screenshot that we had a situation where we had a node named `Skeleton3D4` followed by a node named `Skeleton3D2`. The numbering of these skeletons is based on the order they are encountered in the gltf document as well as 500 lines of an obtuse difficult-to-follow algorithm involving DisjointSet.

Basically, if we ever work on this DisjointSet algorithm, we run the risk of altering the skeleton numbering. By making all Skeleton3Ds named the same, it should reduce the risk that we break compatibility in the future if we edit this algorithm.

**Review by fire — COMMENTED** [review]
We reviewed over discord.

Issues pointed out:

1. I had a rule that gltf nodes and generated nodes should be made unique.
2. We are modifying that rule to be gltf nodes are unique but generated nodes are not made scene unique
3. So skeletons and mesh instances had their uniqueness stripped in the cases below.
4. We believe that skeletons should be counted from the base and not be scene unique as it allows us to work on the node creation algorithm in the future
5. So this means sibling skeletons will be deduplicated by add child.
6. Both mesh and mesh node are going to be the same name

This is blocking a future pr skeleton virtual node case https://github.com/godotengine/godot/pull/71985. So it helps to get this in, and it changes the tree compatibility breaking with editable children.

The biggest benefit of this is that it unblocks the virtual node case by removing nested skeletons.

The skeleton change is not cosmetic but can how animations are stored.

If we decide to switch out the disjoint set algorithm, lyuma strongly believes the skeleton rename will help. But it is easier to make changes based on need and not speculation. So that's negative against changing the skeleton and mesh instance.

We can split the prs, because the nested skeleton is blocking #71985.

**Comment by lyuma** [comment]
Superseded by #72162
